File: apps/web/src/lib/social/profile_query.test.ts
import { test } from 'node:test';
import assert from 'node:assert/strict';
import { extractProfileId } from './profile_query';

const UUID = '3f2504e0-4f89-41d3-9a0c-0305e82c3301';

test('bare uuid is returned lowercased', () => {
	assert.equal(extractProfileId(UUID), UUID);
	assert.equal(extractProfileId(`  ${UUID}  `), UUID);
});

test('an uppercase uuid is normalised to lowercase', () => {
	assert.equal(extractProfileId(UUID.toUpperCase()), UUID);
});

test('a /u/<uuid> path resolves', () => {
	assert.equal(extractProfileId(`/u/${UUID}`), UUID);
});

test('a full profile URL resolves', () => {
	assert.equal(extractProfileId(`https://threkir.com/u/${UUID}`), UUID);
});

test('a /u/<uuid> URL with a query/tab or trailing slash resolves', () => {
	assert.equal(extractProfileId(`https://threkir.com/u/${UUID}?tab=runs`), UUID);
	assert.equal(extractProfileId(`/u/${UUID}/`), UUID);
	assert.equal(extractProfileId(`/u/${UUID}#top`), UUID);
});

test('a plain name is not a profile id', () => {
	assert.equal(extractProfileId('Jane Doe'), null);
	assert.equal(extractProfileId('janedoe'), null);
	assert.equal(extractProfileId('@janedoe'), null);
});

test('empty / whitespace input is null', () => {
	assert.equal(extractProfileId(''), null);
	assert.equal(extractProfileId('   '), null);
});

test('a malformed uuid is not accepted', () => {
	assert.equal(extractProfileId('3f2504e0-4f89-41d3-9a0c'), null);
	assert.equal(extractProfileId('zzzzzzzz-4f89-41d3-9a0c-0305e82c3301'), null);
});

test('a uuid embedded in arbitrary text (not after /u/) is ignored', () => {
	assert.equal(extractProfileId(`hello ${UUID} world`), null);
	assert.equal(extractProfileId(`/runs/${UUID}`), null);
});

File: packages/core_models/test/profile_query_test.dart
import 'package:core_models/core_models.dart';
import 'package:test/test.dart';

// Mirror of apps/web/src/lib/social/profile_query.test.ts — keep in lockstep.
void main() {
  const uuid = '3f2504e0-4f89-41d3-9a0c-0305e82c3301';

  test('bare uuid is returned lowercased', () {
    expect(extractProfileId(uuid), uuid);
    expect(extractProfileId('  $uuid  '), uuid);
  });

  test('an uppercase uuid is normalised to lowercase', () {
    expect(extractProfileId(uuid.toUpperCase()), uuid);
  });

  test('a /u/<uuid> path resolves', () {
    expect(extractProfileId('/u/$uuid'), uuid);
  });

  test('a full profile URL resolves', () {
    expect(extractProfileId('https://threkir.com/u/$uuid'), uuid);
  });

  test('a /u/<uuid> URL with a query/tab or trailing slash resolves', () {
    expect(extractProfileId('https://threkir.com/u/$uuid?tab=runs'), uuid);
    expect(extractProfileId('/u/$uuid/'), uuid);
    expect(extractProfileId('/u/$uuid#top'), uuid);
  });

  test('a plain name is not a profile id', () {
    expect(extractProfileId('Jane Doe'), isNull);
    expect(extractProfileId('janedoe'), isNull);
    expect(extractProfileId('@janedoe'), isNull);
  });

  test('empty / whitespace input is null', () {
    expect(extractProfileId(''), isNull);
    expect(extractProfileId('   '), isNull);
  });

  test('a malformed uuid is not accepted', () {
    expect(extractProfileId('3f2504e0-4f89-41d3-9a0c'), isNull);
    expect(extractProfileId('zzzzzzzz-4f89-41d3-9a0c-0305e82c3301'), isNull);
  });

  test('a uuid embedded in arbitrary text (not after /u/) is ignored', () {
    expect(extractProfileId('hello $uuid world'), isNull);
    expect(extractProfileId('/runs/$uuid'), isNull);
  });
}
